src/composables/useTresContextProvider/index.ts

```diff
@@ -8,18 +8,17 @@ import { extend } from '../../core/catalogue';
 import type { ComputedRef, DeepReadonly, MaybeRef, MaybeRefOrGetter, Ref, ShallowRef } from 'vue';
 
 export type TresContext = {
-  scene: ShallowRef<Scene>;
-  camera: ComputedRef<Camera | undefined>;
-  cameras: DeepReadonly<Ref<Camera[]>>;
+  scene: ShallowRef<Scene>
+  sizes: { height: Ref<number>, width: Ref<number>, aspectRatio: ComputedRef<number> }
+  extend: (objects: any) => void
+  camera: ComputedRef<Camera | undefined>
+  cameras: DeepReadonly<Ref<Camera[]>>
+  controls: Ref<(EventDispatcher & { enabled: boolean }) | null>
   renderer: ShallowRef<WebGLRenderer>
   raycaster: ShallowRef<Raycaster>
-  controls: Ref<(EventDispatcher & { enabled: boolean }) | null>
-  extend: (objects: any) => void
-  addCamera: (camera: Camera) => void;
-  removeCamera: (camera: Camera) => void
-  setCameraActive: (cameraOrUuid: Camera | string) => void;
-
-  sizes: { height: Ref<number>, width: Ref<number>, aspectRatio: ComputedRef<number> }
+  registerCamera: (camera: Camera) => void
```

Comment:
The issue is that whenever you are using the camera from the state and access a property of PerspectiveCamera for example, you would need to force the type, `(camera as PerspectiveCamera).methodOnlyOnPerspective()`

Reply:
I see what you mean. But the same thing would apply for the case when `TresCamera` is used since it is a union type and therefore typescript does not know whether you are using a perspective or othographic camera. If you would like to access `focus` for example, typescript would complain with `focus does not exist on type OrthographicCamera`.

In general, I think the composable should accept any type of camera (including types that might be added in the future).